### PR TITLE
Create an initial spec for Kitchen::Pantry

### PR DESCRIPTION
**Pull request by Stackblocks**
Tests for `.instance`, `.store`, `.get`, `.get!`, `.each` and `.size` methods in the Pantry class.

Re: #48

### CODE REVIEW

File: spec/pantry_spec.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
+require 'spec_helper'
+
+RSpec.describe Kitchen::Pantry do
+  let(:instance) { described_class.new }
+
+  it 'creates a hash' do
```
Comment (jpslav) [inline spec/pantry_spec.rb:6]: the constructor creates a pantry object, not a hash, so need to change this spec name

File: spec/pantry_spec.rb [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    expect(instance).to be_truthy
+  end
+
+  context 'when the pantry has contents' do
+    before { instance.store 'bar', label: 'foo' }
+
+    it 'adds an item to the pantry with the provided label' do
```
Comment (jpslav) [inline spec/pantry_spec.rb:13]: this spec isn't adding an item so need to change the name

File: spec/pantry_spec.rb [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+    it 'gets an item from the pantry' do
+      expect(instance.get(:foo)).to eq('bar')
+    end
+
+    it 'raises if requested item is not in pantry' do
+      expect{ instance.get!('thud') }.to raise_error(
+        Kitchen::RecipeError, "There is no pantry item labeled 'thud'"
+      )
+    end
```
Comment (jpslav) [inline spec/pantry_spec.rb:25]: Would be good to also test the behavior of `get` (not `get!`) when the item isn't present -- it should return `nil`

So you might break it down like 

```ruby
context '#get' do
  it 'returns a present item with either a string or symbol label' do
    expect(instance.get(:foo)).to eq('bar')
    expect(instance.get('foo')).to eq('bar')
  end

  it 'returns nil when an item is not present' do
    expect(instance.get(:blah)).to be_nil
  end
end

context '#get!' do
  it 'returns a present item with either a string or symbol label' do
    expect(instance.get!(:foo)).to eq('bar')
    expect(instance.get!('foo')).to eq('bar')
  end

  it 'raises if the item is not present' do
    # your existing raise spec here
  end
end
```

**Review by jpslav — CHANGES_REQUESTED** [review]

File: spec/pantry_spec.rb (hunk 3 above)
Comment (Stackblocks) [reply spec/pantry_spec.rb:37]: Good point about having positive and negative path tests for both. I've incorporated this and reworked the layout to match 👍

**Review by jpslav — APPROVED** [review]
